docs/concepts/executions.md

```diff
@@ -4,29 +4,36 @@ order: 2
 
 # Execution
 
-An execution is a single run of a flow. An execution in Kestra can exist in many [states](#state).
+An execution is a single run of a flow. An execution has a [state](#state)](#state).
 
 
 ## Task Run
 
-A task run is a task that has been executed. A task run has associated data like Execution Id, State, Start date, End date, etc.
+A task run is a task that has been executed. A task run has associated data like Execution Id, State, Start Date, End Date, etc.
 
 
 ## Attempts 
 
-Each task run will produce one or more attempts. Most task runs will have only one attempt, but you can configure [retries](../developer-guide/retries) for a task. If retries have been configured, they will generate new attempts for each failed task run.
+Each task run can have one or more attempts. Most task runs will have only one attempt, but you can configure [retries](../developer-guide/retries) for a task. 
+If retries have been configured, a task failure will generate new attempts until the retry `maxAttempt` or `maxDuration` threshold was hit.
 
 
 ## Outputs 
 
-Each task can generate output data that other tasks of the current execution can use.
+Each task can generate output data that other tasks of the current flow execution can use.
+
+These outputs can be variables or files that will be stored inside Kestra's [internal storage](../architecture/#the-internal-storage).
+
+Outputs are described on each task documentation page and can be seen in the **Outputs** tab of the **Execution** page.
 
 
 ## Metrics 
 
-Each task can generate metrics data that may be useful in understanding what the current task is doing. For example, `file.size`, `row.count`, etc.
+Each task can generate metrics that may be useful in understanding what the current task is doing. For example: file size, number of rows, etc.
+
+There are two kinds of metrics: duration and counter.
 
-There are two kinds of metrics, duration, and counter. 
+Metrics can be seen in the **Outputs** tab of the **Execution** page.
```

Comment:
```suggestion
Metrics can be seen in the **Outputs** tab of the **Execution** tabs.
```

Reply:
I check and it's the `Executions` page not a tab inside a page.
I use the singular not the plural but it feels more natural as in fact we need to open an execution to have the outputs tab